doc/docs/master/how-tos/use-jupyterhub/using-jupyterhub.md

```diff
@@ -0,0 +1,236 @@
+     transform=python_pachyderm.Transform(cmd=["python3", "/test.py"], image="mytest/testimage"),
+         input=python_pachyderm.Input(pfs=python_pachyderm.PFSInput(glob="/", repo="input_repo")),
+     )
+  client.list_pipeline()
+  ```
+
+  **System response:**
+
+   ```json
+   pipeline_info {
+    pipeline {
+      name: "test"
+    }
+    transform {
+      image: "mytest/testimage"
+      cmd: "python3"
+      cmd: "/test.py"
+    }
+    created_at {
+      seconds: 1578520420
+      nanos: 789017291
+    }
+    version: 1
+    output_branch: "master"
+    resource_requests {
+      memory: "64M"
+    }
+    input {
+      pfs {
+        name: "input_repo"
+        repo: "input_repo"
+        branch: "master"
+        glob: "/"
+      }
+    }
+    cache_size: "64M"
+    salt: "8e57267114c24419a685e250e0fd491b"
+    max_queue_size: 1
+    spec_commit {
+      repo {
+        name: "__spec__"
+      }
+      id: "c83631246a2142cdb93c7a6e4d16fcd2"
+    }
+    datum_tries: 3
+   }
+   ```
+
+For more information, see the
+[OpenCV example](https://github.com/pachyderm/python-pachyderm/blob/master/examples/opencv/opencv.py).
+
+## Create a Repository
+
+To create a repository, run the following code:
+
+```python
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+client.create_repo('<repo-name>')
+client.list_repo()
+```
+
+**Example:**
+
+```python
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+client.create_repo('test')
+client.list_repo()
+```
+
+**System Response:**
+
+```json
+[repo {
+name: "test"
+}
+created {
+  seconds: 1576869000
+  nanos: 886123695
+}
+auth_info {
+  access_level: OWNER
+}
+]
+```
+
+## Add Files to a Repository
+
+To add a file to a repository, run the following code:
+
+```python
+client.put_file_url("<repo-name>/<branch>", "<filename>", "<path-to-file>")
+```
+
+**Example:**
+
+```python
+client.put_file_url("images/master", "46Q8nDz.jpg", "http://imgur.com/46Q8nDz.jpg")
+```
+
+## Delete a Repository
+
+To delete a repository, run the following code:
+
+```python
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+client.delete_repo('test')
+client.list_repo()
+```
+
+**System Response:**
+
+```json
+[]
+```
+
+## Update Your Pipeline
+
+When you need to update your pipeline, you can do so directly in the
+JupyterHub UI by modifying the corresponding notebook and running it
+again. If you use the `create_python_pipeline` function that uses the
+code stored in a local directory, you can update the pipeline directly
+in the JupyterHub UI by adding the `update=True` parameter to your
+code into a new Jupyter notebook cell and running it.
+
+**Example:**
+
+```python hl_lines="8"
+import os
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+python_pachyderm.create_python_pipeline(
+ client,
+ "./edges",
+ python_pachyderm.Input(pfs=python_pachyderm.PFSInput(glob="/*", repo="images")),
+ update=True
+)
+```
```

Comment:
this code sample uses one space tabs, and the code samples above use a mix of 2 and 4 space tabs

Reply:
added 4 spaces.